pkg/api/pod/util.go

```diff
@@ -345,6 +345,59 @@ func usesMultipleHugePageResources(podSpec *api.PodSpec) bool {
 	return len(hugePageResources) > 1
 }
 
+func checkContainerUseIndivisibleHugePagesValues(container api.Container) bool {
+	for resourceName, quantity := range container.Resources.Limits {
+		if helper.IsHugePageResourceName(resourceName) {
+			if !helper.IsHugePageResourceValueDivisible(resourceName, quantity) {
+				return true
+			}
+		}
+	}
+
+	for resourceName, quantity := range container.Resources.Requests {
+		if helper.IsHugePageResourceName(resourceName) {
+			if !helper.IsHugePageResourceValueDivisible(resourceName, quantity) {
+				return true
+			}
+		}
+	}
+
+	return false
+}
+
+// usesIndivisibleHugePagesValues returns true if the one of the containers uses non-integer multiple
+// of huge page unit size
+func usesIndivisibleHugePagesValues(podSpec *api.PodSpec) bool {
+	for _, ctr := range podSpec.Containers {
+		if checkContainerUseIndivisibleHugePagesValues(ctr) {
+			return true
+		}
+	}
+
+	for _, ctr := range podSpec.InitContainers {
+		if checkContainerUseIndivisibleHugePagesValues(ctr) {
+			return true
+		}
+
+	}
+
+	for i := range podSpec.EphemeralContainers {
+		if checkContainerUseIndivisibleHugePagesValues((api.Container)(podSpec.EphemeralContainers[i].EphemeralContainerCommon)) {
+			return true
+		}
+	}
```

Comment:
VisitContainers will iterate over all container types for you:

```go
foundIndivisibleHugePagesValue := false
VisitContainers(podSpec, AllContainers, func(ctr *api.Container, containerType ContainerType) (shouldContinue bool) {
  if checkContainerUseIndivisibleHugePagesValues(ctr) {
    foundIndivisibleHugePagesValue = true
  }
  return !foundIndivisibleHugePagesValue // continue visiting if we haven't seen an invalid value yet
})
if foundIndivisibleHugePagesValue {
  return true
}
```

Reply:
thanks for suggestion. I have updated the code according to the comment. Please double-check.